### PR TITLE
Add code formatting support

### PR DESCRIPTION
**Pull request by chainchad**
- Creates new `ci-prettier` action
- Uses new `ci-prettier` action for PR and main push workflows
- Adds lefthook for pre-commit checks with prettier
- Format all relevant files with prettier

### CODE REVIEW

File: .changeset/wicked-worms-change.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"ci-prettier": patch
```
Comment (momentmaker) [inline .changeset/wicked-worms-change.md:2]: should this be `minor`?

Comment (chainchad) [reply .changeset/wicked-worms-change.md]: Thanks, updated in 8f164b2f1934e17f28d953350aaea6d6d2efcb2a

File: package.json [hunk 2]
```diff
@@ -6,8 +6,8 @@
     "pnpm": "9"
   },
   "scripts": {
-    "prettier:check": "prettier --check **/*.{json,md} --ignore-path=.prettierignore",
-    "prettier:write": "prettier --write **/*.{json,md} --ignore-path=.prettierignore",
+    "prettier:check": "prettier --check \"**/*.{json,md,js,ts,yml,yaml}\" --ignore-path .prettierignore",
```
Comment (momentmaker) [inline package.json:9]: why was the `=` removed here?

Comment (chainchad) [reply package.json]: Unintentional. Added back in b85afc670cbe22a42c4db9977441f37525514ca9.

**Review by momentmaker — APPROVED** [review]

File: lefthook.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+pre-commit:
+  parallel: true
+  commands:
+    prettier:
+      # run: pnpm run prettier:check
+      description: "Check for prettier formatting"
+      run: npx prettier --check {staged_files} --ignore-path=.prettierignore
```
Comment (erikburt) [inline lefthook.yml:7]: should this be using `pnpm` ?

**Review by erikburt — COMMENTED** [review]
Maybe make mention of `lefthook` and how to install the git-hook in the `README`?

File: lefthook.yml (hunk 3 above)
Comment (chainchad) [reply lefthook.yml]: Good catch. Updated in 929f32be967b63aa49e24ed7b3bcb81240a346b9.

**Review by erikburt — APPROVED** [review]

**Pull request merged**